baybe/parameters/substance.py

```diff
@@ -58,10 +57,28 @@ class SubstanceParameter(DiscreteParameter):
     """
 
     encoding: SubstanceEncoding = field(
-        default=SubstanceEncoding.MORDRED, converter=SubstanceEncoding
+        default=SubstanceEncoding.MordredFingerprint,
+        converter=lambda x: (
+            # Passed enum
+            x
+            if isinstance(x, SubstanceEncoding)
+            # Passed enum name
+            else (
+                SubstanceEncoding[x]
+                if x in SubstanceEncoding.__members__
+                # Passed enum value
+                else SubstanceEncoding(x)
+            )
+        ),
```

Comment:
Haven't yet quite looked in detail at the logic, but this by far the longest converter I've seen 😄 There must be a more elegant / readable way

Reply:
I think the others are shorter for enums as names equal values, but this enum has currently different names and values, which requires these checks. But feel free to suggest how to adapt

Reply:
perhaps have a try to let chatgpt compress this to the max, but overall I dont think this is terribly long (the situation is simply more complex as with converters used elsewhere with enums)
other than that feel free to resolve

Reply:
not relevant any more as enum was changed